cpp/src/arrow/util/simd.h

```diff
@@ -17,6 +17,24 @@
 
 #pragma once
 
+#ifdef _MSC_VER
+// MSVC x86_64/arm64
+
+#if defined(_M_AMD64) || defined(_M_X64)
+#include <intrin.h>
+#elif defined(_M_ARM64)
+#include <arm64_neon.h>
+#endif
+
+#else
+// gcc/clang (possibly others)
+
```

Comment:
Just an FYI in https://github.com/apache/arrow/pull/6985 relies on BMI2 do you think adding that here would be appropriate?

Reply:
yes, I think so